src/expressions/functions/builtInFunctions_maps.ts

```diff
@@ -231,8 +231,8 @@ export default {
 			namespaceURI: MAP_NAMESPACE_URI,
 			localName: 'for-each',
 			// TODO: reimplement type checking by parsing the types
-			// argumentTypes: ['map(*)', 'function(xs:anyAtomicType, item()*) as item()*'],
-			argumentTypes: ['map(*)', 'function(*)'],
+			// argumentTypes: ['map(*)', '(xs:anyAtomicType, item()*) as item()*'],
```

Comment:
Not sure if this one was intended - looks like this was an XPath function type?

Reply:
It was intended. Ideally, function types would be supported in the form of the comment above. However, there is no support yet.